src/3rd_party/pathie-cpp/src/pathie.cpp

```diff
@@ -143,7 +143,7 @@ std::string Pathie::convert_encodings(const char* from_encoding, const char* to_
     errno  = 0;
     errsav = 0;
 
-#ifdef BSD
+#if defined(BSD) && ! defined(__APPLE__) //Since MacOS evolved from BSD, it is captured here but the iconv on macos behaves differently
```

Comment:
Is this distinction needed at all? Can we not always use the version with the const_cast, and leave the comment in to explain why? It seems the other arguments are all the same?

Reply:
I have not tried it, maybe. I will try tomorrow.

Reply:
No, the BSD codepath doesn't compile on Linux. We can't merge the two, and since BSD captures MAC, we have to add the exception for APPLE in the ifdef

Reply:
Can't one always pass a non-const pointer where a const pointer is expected?

Reply:
I don't have access to a BSD machine to test if this codepath is absolutely necessary for compilation in BSD. The upstream author of `pathie-cpp` deemed it was necessary and coded it in this way. This change was accepted by him when I added mac support upstream. I don't see a reason to deviate from it. If you want, I can remve BSD support altogether.